pkg/utils/transform/transform.go

```diff
@@ -48,3 +49,54 @@ func InjectNamespaceConditional(preserveNamespace string, targetNamespace string
 		return tf(u)
 	}
 }
+
+func ReplaceKind(fromKind, toKind string) mf.Transformer {
+	return func(u *unstructured.Unstructured) error {
+		kind := u.GetKind()
+
+		if kind != fromKind {
+			return nil
+		}
+		err := unstructured.SetNestedField(u.Object, toKind, "kind")
+		if err != nil {
+			return fmt.Errorf("could change resource KIND, %q", err)
+		}
+		return nil
+	}
+}
+
+func InjectLabel(key, value string, overwrite bool, kinds ...string) mf.Transformer {
```

Comment:
may want to return an error if `kinds` length is 0

Reply:
my reasoning is, `if len(kinds)==0` that should mean `any` resource
I will add better comments

Reply:
```
//InjectLabel adds label key:value to a resource
// overwritePolicy decides whehter to overwite an already existing label
// []kinds specify the Kinds on which the label should be applied
// if len(kinds) = 0, label will be apllied to all/any resources irrespective of its Kind
```